cli/flox/src/commands/generations/rollback.rs

```diff
@@ -92,12 +100,22 @@ mod tests {
     #[test]
     fn rollback_rolls_back_to_newer_generation_if_previously_active() {
         // e.g. rolled back once 3->2
-        let metadata = mock_generations(2.into());
+        // last created active
+        let mut metadata = AllGenerationsMetadata::default();
+        metadata.add_generation(default_add_generation_options());
+        metadata.add_generation(default_add_generation_options());
+        let (third_generation, ..) = metadata.add_generation(default_add_generation_options());
+        metadata
+            .switch_generation(default_switch_generation_options(
+                determine_previous_generation(&metadata).unwrap().0,
+            ))
+            .unwrap();
+
```

Comment:
(praise) Nice that the mock is simpler and this exercises cycling through the real states 👍🏻